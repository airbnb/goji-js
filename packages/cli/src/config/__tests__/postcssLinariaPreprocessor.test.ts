import postcss from 'postcss';

const transform = async (css: string) => {
  // eslint-disable-next-line global-require
  const result = await postcss([require('../postcssLinariaPreprocessor')()]).process(css, {
    from: '/path/to/file.css',
  });
  return result.css;
};

describe('postcss-linaria-preprocessor', () => {
  test(':global() selector', () => {
    const css = `
      .a {
        color: red;
        :global() {
          page {
            width: 50vw;
          }
        }
      }
    `;
    expect(transform(css)).resolves.toMatchSnapshot();
  });

  test('keyframes rename', () => {
    const css = `
      .a {
        animation: 1s ease 1 backwards normal zoomIn;
        @keyframes zoomIn {
          from {
            opacity: 0;
            transform: scale(0.94);
          }

          50% {
            opacity: var(--opacity);
          }

          to {
            transform: scale(var(--scale));
          }
        }
      }
    `;
    expect(transform(css)).resolves.toMatchSnapshot();
  });

  test('escape breaking control characters', () => {
    const css = `
      .a {
        content: '\feff';
      }
    `;
    expect(transform(css)).resolves.toMatchSnapshot();
  });
});
