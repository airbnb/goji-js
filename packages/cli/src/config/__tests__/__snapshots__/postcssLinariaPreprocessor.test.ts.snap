// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-linaria-preprocessor :global() selector 1`] = `
"
      .a {
        color: red
      }
          page {
            width: 50vw;
          }
    "
`;

exports[`postcss-linaria-preprocessor escape breaking control characters 1`] = `
"
      .a {
        content: '\\\\feff';
      }
    "
`;

exports[`postcss-linaria-preprocessor keyframes rename 1`] = `
"
      .a {
        animation: 1s ease 1 backwards normal zoomIn-a;
        @keyframes zoomIn-a {
          from {
            opacity: 0;
            transform: scale(0.94);
          }

          50% {
            opacity: var(--opacity);
          }

          to {
            transform: scale(var(--scale));
          }
        }
      }
    "
`;
